### DISCUSSION TITLE
Sampling the "table of elements" as an input for a PINN

### DISCUSSION THREAD

**Discussion by valderrama-juan** [lululxvi/deepxde#1690] [Q&A]
Hi,

I am currently training a PINN that implements the atomic number Z and the atomic mass number m as inputs of the network. In short, the PINN models the diffusion of a solid "heavy" into a gaseous "light" material, specifically into deuterium (Z=1, m=2). I would like to train the PINN to learn solutions for various heavy materials, from carbon (Z=6, m=12) to gold (Z=79, m=197).

The way I am doing this right now is by setting minimum and maximum bounds to both my Z and m values in the following way:

```
Z_I_Min = 1
Z_I_Max = 100
m_I_Min = 1
m_I_Max = 200
```

These are then added to the geometry of the problem in the following manner:

`geom = dde.geometry.Hypercube([xMin, yMin, Z_I_Min, m_I_Min], [xMax, yMax, Z_I_Max, m_I_Max])`

(x and y are the spatial inputs). My issue is that this "wastes" a lot of points on non-existing and non-physical elements, such as (Z=1, A=200) or (Z=100, A=1). Ideally, I would like to train one network for various elements, but since NNs are continuous, I'm not sure how to sample the training points into a distribution that somewhat follows the A~2*Z shape of the elements curve.

**Comment by lululxvi** [comment]
Do you mean you only want to sample some integer points for Z and m? If so, then you have to generate the training points by yourself, or modify the sampling code in DeepXDE.
